src/lib/utils/ghash/ghash.cpp

```diff
@@ -178,7 +178,7 @@ void GHASH::add_final_block(secure_vector<uint8_t>& hash, size_t ad_len, size_t
    * and the length of the AD is probably not sensitive either.
    */
    uint8_t final_block[GCM_BS];
-   store_be<uint64_t>(final_block, 8 * ad_len, 8 * text_len);
+   store_be(final_block, 8 * static_cast<uint64_t>(ad_len), 8 * static_cast<uint64_t>(text_len));
```

Comment:
`store_be<>` now takes a variadic number of (same-length) integers. `ad_len` and `text_len` are `size_t` (which is 32-bit on 32 bit machines). Therefore, we need to specifically cast those to `uint64_t`.

Reply:
This looks fine, but it might be a bit cleaner to do something like

```
const uint64_t ad_bits = 8 * static_cast<uint64_t>(ad_len);
// similar for text_len
store_be(final_block, ad_bits, text_bits);
```

Reply:
aye aye.